bee-node/bee-node/src/tools/convert.rs

```
@@ -0,0 +1,105 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_block::address::{Address, Ed25519Address};
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+use structopt::StructOpt;
+use thiserror::Error;
+
+const BECH32_HRP: &str = "iota";
+
+#[derive(Clone, Debug, Error)]
+pub enum ConvertError {
+    #[error("invalid Bech32 address length")]
+    InvalidAddressLength(),
+    #[error("Invalid Address")]
+    InvalidAddress(),
+}
+
+#[derive(Clone, Debug, StructOpt)]
+pub enum ConvertTool {
+    /// convert Bech32 address to hex encoding
+    Bech32ToHex { bech32: String },
+    /// convert hex encoding to Bech32 address
+    HexToBech32 { hex: String },
+    /// convert a hex encoded public key to Bech32 address
+    HexPubkeyToBech32 { pubkey: String },
+}
+
+pub fn exec(tool: &ConvertTool) -> Result<(), ConvertError> {
+    match tool {
+        ConvertTool::Bech32ToHex { bech32 } => {
+            let hex = bech32_to_hex(bech32.as_str()).unwrap();
```

You can't unwrap, you need to propagate the error